snapshot/snapshot.go

```diff
@@ -914,6 +889,11 @@ func (o *snapshotter) getCleanupDirectories(ctx context.Context) ([]string, erro
 		if _, ok := ids[d]; ok {
 			continue
 		}
+		// Skip temp snapshot directories
+		if strings.HasPrefix(d, "new-") {
```

Comment:
Is there a reason why we should keep the temp directories?

Reply:
snapshotter::createSnapshot() create a temporary directory named as `new-xxx` under the cache directory, which may be deleted by the gc thread, so we should protect it.

Reply:
What gc thread? Do you mean containerd GC routines?

Reply:
Aha, you are right. The should keep recent temp files and remove old temp files. So change it to:
```
		// Temporary files may still be in use, so only delete temp files elder than one day.
		if strings.HasPrefix(d, "new-") {
			if fi, err := os.Stat(o.snapshotDir(d)); err == nil {
				if fi.ModTime().Add(24 * time.Hour).After(time.Now()) {
					continue
				}
			}
		}
```

Reply:
It seems unnecessary to handle this race. containerd snapshot gc and prepare have rwlock protection.

https://github.com/containerd/containerd/blob/1f54e8fb21ff565bc240a1a9ca29e4566e4ce617/metadata/snapshot.go#L300

https://github.com/containerd/containerd/blob/1f54e8fb21ff565bc240a1a9ca29e4566e4ce617/metadata/snapshot.go#LL831C12-L831C12

Reply:
Thanks, I will remove this patch.